src/SdkCommon/ClientRuntime/ClientRuntime/RetryDelegatingHandler.cs

```diff
@@ -139,51 +133,15 @@ public int EventCallbackCount
 
                 return responseMessage;
             }
-            catch(Exception ex)
-            {
-                if (responseMessage != null)
-                {
-                    return responseMessage;
-                }
-                else
-                {
-                    throw ex;
-                }
-            }
-            finally
+            catch (Exception)
             {
-                if (Retrying != null)
+                if (responseMessage != null || lastErrorResponseMessage != null)
                 {
-                    foreach (EventHandler<RetryingEventArgs> d in Retrying.GetInvocationList())
-                    {
-                        Retrying -= d;
-                    }
+                    return responseMessage ?? lastErrorResponseMessage;
                 }
+                
+                throw;
             }
         }
-
-        /// <summary>
-        /// Function that makes sure we subscribe once to RetryPolicy Retrying event once
-        /// </summary>
-        private void SubscribeRetryPolicyRetryingEvent()
-        {
-            if(_retryPolicyRetryingEventSubscribed == false)
-            {
-                RetryPolicy.Retrying += (sender, args) =>
-                {
-                    if (Retrying != null)
-                    {
-                        Retrying(sender, args);
-                    }
-                };
-
-                _retryPolicyRetryingEventSubscribed = true;
-            }
-        }
-
-        /// <summary>
-        /// An instance of a callback delegate that will be invoked whenever a retry condition is encountered.
-        /// </summary>
-        private event EventHandler<RetryingEventArgs> Retrying;
```

Comment:
You cannot remove the event that is part of our existing extensibility provided via RetryPolicy.
This event is wired up via RetryPolicy that customers can set according to their needs.
Please have all existing functionality back and then have the required code change.

Reply:
The EventHandler being removed here is private and does not impact the existing extensibility provided by the RetryPolicy.